epidose/device/util.sh

```diff
@@ -186,6 +195,19 @@ get_new_filter()
   fi
 }
 
+# Check and update software if necessary
+# preconditions: WiFi should be turned on
+check_for_updates()
+{
+  log "Checking for updates"
+  if ! git -C "$LOCAL_REPO" remote show origin | grep "up to date" ; then
+    log "Update is needed"
+    git -C "$LOCAL_REPO" pull "$REMOTE_REPO" production && kill -9 "$(cat "$SLEEP_UPDATE_SOFTWARE_PID")"
```

Comment:
What is the benefit of this complicated (and error-prone) scheme of two deamons with the one waking up the other? Why not do the updates inline here?